pkg/controller/command/verifiable/command.go

```diff
@@ -763,11 +761,43 @@ func prepareOpts(opts *ProofOptions, didDoc *did.Doc) (*ProofOptions, error) {
 		opts = &ProofOptions{}
 	}
 
+	opts.proofPurpose = "authentication"
+
+	if opts.PrivateKey != "" && opts.VerificationMethod == "" {
+		return nil, fmt.Errorf("verification method matching given private key is not provided")
+	}
+
+	authVMs := didDoc.VerificationMethods(did.Authentication)[did.Authentication]
+
+	vmMatched := opts.VerificationMethod == ""
+
+	for _, vm := range authVMs {
+		if opts.VerificationMethod != "" {
+			// if verification method is provided as an option, then validate if it belongs to 'authentication'
+			if opts.VerificationMethod == vm.PublicKey.ID {
+				vmMatched = true
+				break
+			}
+
+			continue
+		} else {
+			// by default first authentication public key
+			opts.VerificationMethod = vm.PublicKey.ID
+			break
+		}
+	}
+
+	if !vmMatched {
+		return nil, fmt.Errorf("unable to find matching 'authentication' key IDs for given verification method")
+	}
+
+	// this is the fallback logic kept for DIDs not having authentication method
+	// TODO to be removed [Issue #1693]
```

Comment:
we still need this?

Reply:
I didn't want to break any existing features, will be removed in next PR.